### ISSUE TITLE
Fatal error: resources.lib.service

### ISSUE THREAD

**Issue by Fred-Vatin** [razzeee/script.trakt#390]
script version | Module version | Kodi | OS | 
---------------| --- | ---- | ----
3.1.12 | 2.14.1 | 17.3 | Win 10 x64 pro (french) 14393 

## Steps to reproduce

1. Trakt settings: disable TV episode scrobble.
2. play an episode from a TV show

## Results

If trakt setting is to rate episode after watch, [here](https://paste.ubuntu.com/25268557/) is the log with fatal error: `AttributeError: 'NoneType' object has no attribute 'to_dict'` (line 279 & 337).

At the end of episode, trakt doesn't offer to rate it.

If trakt setting is to NOT rate episode after watch, [here](https://paste.ubuntu.com/25268590/) is the log with another fatal error: `Error Type: <type 'KeyError'>` (line 270).

[Here](https://gist.github.com/freMea/868c4eb75a2e7d17e2fd00323bb739a0) is my trakt settings.

## No issue cases

* when playing a movie.
* if Trakt settings: TV episode scrobble is ON → no issue and it asks me to rate at the end of episode.

Maybe related (or not at all) with the fact my database is filled with the NFO scrapper. See #387

**Comment by razzeee** [comment]
- First error is due to trakt not beeing able to find a tvshow with the id we hand it. 

- Secound error looks like it's going to happen for the above case also, but you don't reach it (lucky you). Will probably also have the same root cause.

So we will need your nfo again :)
And please check the ids vs trakt

**Comment by Fred-Vatin** [comment]
@Razzeee 

For the NFO, you can check the same I posted in the #387, that is to say [here](https://gist.github.com/freMea/f1c48af895c9f30cbadb7fcb42cf8fc0). Issue exists for every show and every episode. There is NO issue with sync, all my show that are on my Kodi DB are on my trakt DB.

When scobble episode is ON, there is no issue, I get the notification it's ok. Same with the rating I send at the end of episode. It seems there is no issue in this context for trakt to find my tvshow and it works flawlessly.

Fatal errors only pop up in the log if scobble episode is OFF and it breaks the rating feature. I think it is the key of the issue but I don't know why.

**Comment by razzeee** [comment]
Yeah, it's a path that's not very well tested. At least not from me, not sure when I will have time for that.
